### PR TITLE
Add sticky navbar functionality

### LINKED ISSUE #232: Make frontend Header sticky on the page
**Is your feature request related to a problem? Please describe.**
When we scroll on the page with many items we have to scroll all the way up to get to a different page (e.g. Projects -> Chapters).

**Describe the solution you'd like**
Making Header component sticky on the page will help with navigation between pages.

### PR DESCRIPTION
**Pull request by nitinawari**
Fixed #232 
**Checkpoints**
 - Added a sticky navbar.
 - Ensured that when users switch pages, the page scrolls to the top automatically.
 - Added test cases for all functionalities.

image

**Demo*

https://github.com/user-attachments/assets/aecf23a7-e1ec-4a2d-9bb9-aace92187da1

### CODE REVIEW

File: frontend/src/components/Header.tsx
Comment (kasya) [inline frontend/src/components/Header.tsx:1]: It works great on the regular page. However on the mobile view the header is not really sticky. 
I'm not talking about the responsiveness, we are covering that in a different task, but the stickiness is not really applying properly. Can we make sure it works as expected?

https://github.com/user-attachments/assets/d8123f6a-b728-42a6-b433-c46affa36255

Comment (nitinawari) [reply frontend/src/components/Header.tsx:1]: Ok i will resolve

Comment (nitinawari) [reply frontend/src/components/Header.tsx:1]: @kasya
I attempted to solve it, To solve this issue, which is caused by the lack of header responsiveness, we need to address header responsiveness first to successfully implement a sticky header on mobile. What should we do ?

Comment (kasya) [reply frontend/src/components/Header.tsx:1]: @nitinawari gotcha! I know we already have a task in progress for that. In that case let's ignore header not being perfectly sticky for now and revisit this after all responsiveness issues have been addressed. 👌🏼

**Review by kasya — APPROVED** [review]
Nice update! Thanks!

**Comment by nitinawari** [comment]
@kasya  @arkid15r  can we merge it !

**Pull request merged**